src/main/java/net/countercraft/movecraft/combat/movecraftcombat/listener/IgniteListener.java

```diff
@@ -0,0 +1,86 @@
+import net.countercraft.movecraft.craft.CraftManager;
+import net.countercraft.movecraft.utils.MathUtils;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.WorldGuard6Utils;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.LegacyUtils;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockIgniteEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class IgniteListener implements Listener {
+    @EventHandler(priority = EventPriority.LOW)
+    public void onBlockIgnite(BlockIgniteEvent event) {
+        if (event.isCancelled()) {
+            return;
+        }
+        final Craft adjacentCraft = adjacentCraft(event.getBlock().getLocation());
+
+
+        // replace blocks with fire occasionally, to prevent fast craft from simply ignoring fire
+        if (Config.EnableFireballPenetration && event.getCause() == BlockIgniteEvent.IgniteCause.FIREBALL) {
+            Block testBlock = event.getBlock().getRelative(-1, 0, 0);
+            if (!testBlock.getType().isBurnable())
+                testBlock = event.getBlock().getRelative(1, 0, 0);
+            if (!testBlock.getType().isBurnable())
+                testBlock = event.getBlock().getRelative(0, 0, -1);
+            if (!testBlock.getType().isBurnable())
+                testBlock = event.getBlock().getRelative(0, 0, 1);
+
+            if (!testBlock.getType().isBurnable()) {
+                return;
+            }
+
+            // check to see if fire spread is allowed, don't check if worldguard integration is not enabled
+            if(Movecraft.getInstance().getWorldGuardPlugin() != null && (Settings.WorldGuardBlockMoveOnBuildPerm ||Settings.WorldGuardBlockSinkOnPVPPerm)) {
+                if (LegacyUtils.getInstance().isLegacy()) {
+                    if (!WorldGuard6Utils.locationAllowsFireSpread(event.getBlock().getLocation())) {
+                        return;
+                    }
+                } else {
+                    RegionManager manager = WorldGuard.getInstance().getPlatform().getRegionContainer().get(new BukkitWorld(testBlock.getWorld()));
+                    ApplicableRegionSet set = manager.getApplicableRegions(BlockVector3.at(testBlock.getX(), testBlock.getY(), testBlock.getZ()));
+                    for (ProtectedRegion region : set) {
+                        if (region.getFlag(Flags.FIRE_SPREAD) == StateFlag.State.DENY) {
+                            return;
+                        }
+                    }
+                }
+            }
+            testBlock.setType(Material.AIR);
+        }
+        if (adjacentCraft != null && Config.AddFiresToHitbox) { //else if
+            adjacentCraft.getHitBox().add(MathUtils.bukkit2MovecraftLoc(event.getBlock().getLocation()));
```

Comment:
locIsNearCraftFast will return true if a block is within 3 blocks of a craft.  In the odd case that a fire is started in air next to a craft (or say, on a dock next to it), this would currently add that to the craft's hitbox.  I am uncertain of a better function to use for this.